File: igtampe.cdbfs.frontend/src/model/CdbfsFolder.ts
import Editable from "./Editable";

export default class CdbfsFolder extends Editable{
    public constructor(
        public id : number,
        public driveId : number,
        public name:string,
        public fileCount:number,
        public folderCount:number,
        public size:number,
        createTs : string,createUserId: string,updateTs?: string,updateUserId?: string,
        public parentFolder?: number,
    ){
        super(createTs,createUserId,updateTs,updateUserId)
    }
}

File: Igtampe.CDBFS.Common/CdbfsFolder.cs
namespace Igtampe.CDBFS.Common {
    public class CdbfsFolder : Editable{

        public int DriveId { get; set; } = 0;
        public int Id { get; set; } = 0;
        public int? ParentFolder { get; set; } = null;
        public string Name { get; set; } = "";

        public int FileCount { get; set; } = 0;
        public int FolderCount { get; set; } = 0;
        public long Size { get; set; } = 0;
        
    }
}
